delivery_carrier_deposit/wizard/deposit.py

```diff
@@ -40,23 +40,33 @@ def _get_carrier_type_selection(self):
         required=True, help="Carrier type (combines several delivery "
         "methods). Make sure that the option 'Deposit Slip' is checked on "
         "the delivery methods that have this carrier type.")
+    picking_type_id = fields.Many2one(
+        comodel_name='stock.picking.type',
+        string='Picking Type',
+        required=True)
 
     @api.model
     def _prepare_deposit_slip(self):
         return {
             'carrier_type': self.carrier_type,
             'company_id': self.env.user.company_id.id,
+            'picking_type_id': self.picking_type_id.id,
             }
 
+    @api.model
+    def _get_deposit_pickings(self):
+        return self.env['stock.picking'].search([
```

Comment:
That's right but only change the domain is very restrictive in my opinion.

You could want to get pickings with very specific search from sale_order for instance.

And I see two case of overriding

1. for a **carrier** in that case you want to test depending on the carrier_type and you have your own domain 

`if self.carrier_type  == 'carrier_type':
     domain = []
 else:
     super ..`

Because you maybe don't want the other carrier addition in the domain

2. a specific domain for a customer so you will maybe not even call the super() to make your specific search

In the two case, changing the domain or directly searching the pickings ends with the same amount of searchs but more flexible to return pickings instead of domain.

This is what I think but if you think it's more safe to only return the domain, I agree to change for your idea.